lib/reevoo-ping.js

```diff
@@ -12,13 +12,20 @@ import Badge                  from './events/badge';
 import Conversion             from './events/conversion';
 
 class Client {
-  constructor(appId, opts = {}, trackerConfig = {}) {
+  constructor(appId, opts = {}, trackerConfig = {}, win = window) {
     if (!appId) throw new Error('appId argument is required');
 
     // Initialise the tracker
     const snowplowTrackerConfig = { ...defaultSnowplowConfig, ...trackerConfig, appId };
     snowplow('newTracker', appId, COLLECTOR_URI, snowplowTrackerConfig);
 
+    // Snowplow collector cannot handle URLs with characters violating RFC 2396, if customer URL contains some of
+    // them we set our custom encoded version to be tracked.
+    const encodedURI = encodeURI(win.location.href).replace(/!/g, '%21');
```

Comment:
Out of interest, why does `!` need to be replaced, does it not get rewritten by `encodeURI`?  Why is it unique; it's a bit weird....

Minor point - can we have variable a different name to the method name?  Not massively important, but just a suggestion :)